catroid/src/main/java/org/catrobat/catroid/formulaeditor/FormulaElement.java

```diff
@@ -214,16 +208,36 @@ public void updateCollisionFormula(String oldName, String newName) {
 		if (rightChild != null) {
 			rightChild.updateCollisionFormula(oldName, newName);
 		}
-		if (type == ElementType.COLLISION_FORMULA && value.contains(oldName)) {
-			String collisionTag = CatroidApplication.getAppContext().getString(R.string
-					.formula_editor_function_collision);
-			String firstSprite = value.substring(0, value.indexOf(collisionTag) - 1);
-			String secondSprite = value.substring(value.indexOf(collisionTag) + collisionTag.length() + 1, value.length());
-			if (firstSprite.equals(oldName)) {
-				value = newName + " " + collisionTag + " " + secondSprite;
-			} else if (secondSprite.equals(oldName)) {
-				value = firstSprite + " " + collisionTag + " " + newName;
+		if (type == ElementType.COLLISION_FORMULA && value.equals(oldName)) {
+			value = newName;
+		}
+	}
+
+	public void updateCollisionFormulaToNewVersion() {
+		if (leftChild != null) {
+			leftChild.updateCollisionFormulaToNewVersion();
+		}
+		if (rightChild != null) {
+			rightChild.updateCollisionFormulaToNewVersion();
+		}
+		if (type == ElementType.COLLISION_FORMULA) {
+			int indexOfSpriteInFormula = -1;
+			for (Scene scene : ProjectManager.getInstance().getCurrentProject().getSceneList()) {
+				for (Sprite sprite : scene.getSpriteList()) {
+					int index = value.indexOf(sprite.getName());
+					if (index > indexOfSpriteInFormula) {
+						indexOfSpriteInFormula = index;
+					}
+				}
+			}
```

Comment:
Consider a value of `a touches ab` and three sprites (`a`, `b`, `ab`). The parsed secondSpriteName will be `b` instead of `ab`.
If you pass the firstSprite as parameter, you are able to parse the secondSpriteName reliably (at least if a translation of `touches` won't screw up your parsing :wink:).